File: Harpia-Health-Analysis-WS/src/main/java/com/harpia/HarpiaHealthAnalysisWS/model/enums/EnumUserRole.java
package com.harpia.HarpiaHealthAnalysisWS.model.enums;

public enum EnumUserRole {
    ADMIN(1, "Admin"), DOCTOR(2, "DOCTOR"),/*HEALTHCARE_PERSONAL(2, "Healthcare_Personal")*/ PATIENT(3, "Patient");

    private final int id;
    private final String role;

    EnumUserRole(int id, String role) {
        this.id = id;
        this.role = role;
    }

    public int getId() {
        return this.id;
    }

    public String getName() {
        return this.role;
    }
}

File: flutter-mobil/lib/model/userrole/EnumUserRole.dart
enum EnumUserRole {
  ADMIN(roleId: 1, roleName: "Admin"),
  DOCTOR(roleId: 2, roleName: "Doctor"),
  PATIENT(roleId: 3, roleName: "Patient");

  final int roleId;
  final String roleName;

  const EnumUserRole({required this.roleId, required this.roleName});

  static String getRoleName(int roleId) {
    switch (roleId) {
      case 1:
        return EnumUserRole.ADMIN.roleName;
      case 2:
        return EnumUserRole.DOCTOR.roleName;
      case 3:
        return EnumUserRole.PATIENT.roleName;
      default:
        throw ArgumentError("Invalid roleId");
    }
  }
}
